app/menu.js

```diff
@@ -625,35 +570,75 @@ const createMenu = (CommonMenu) => {
   const oldMenu = appMenu
   appMenu = Menu.buildFromTemplate(template)
   Menu.setApplicationMenu(appMenu)
-  oldMenu.destroy()
+  if (oldMenu) {
+    oldMenu.destroy()
+  }
+}
+
+const updateMenu = (CommonMenu, appState, windowState) => {
+  const updated = menuUtil.checkForUpdate(appState, windowState)
+  if (updated.nothingUpdated) {
+    return
+  }
+
+  // Only update menu when necessary
+
+  if (updated.settings || updated.closedFrames) {
+    console.log('rebuilding history menu')
```

Comment:
logs from debugging? Do we need/want these? We really need to add our own log method with more control over output granularity